python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/resources.py

```diff
@@ -1,20 +1,90 @@
-from dagster import resource
+from contextlib import contextmanager
+from typing import Any, Iterator, Optional
+
+from dagster import ConfigurableResource, IAttachDifferentObjectToOpContext, resource
 from google.cloud import bigquery
+from pydantic import Field
 
-from .configs import bq_resource_config
 from .utils import setup_gcp_creds
 
 
+class BigQueryResource(ConfigurableResource, IAttachDifferentObjectToOpContext):
+    """Resource for interacting with Google BigQuery.
+
+    Examples:
+        .. code-block:: python
+
+            from dagster import Definitions, asset
+            from dagster_gcp import BigQueryResource
+
+            @asset
+            def my_table(bigquery: BigQueryResource):
+                with bigquery.get_client() as client:
+                    client.query("SELECT * FROM my_dataset.my_table")
+
+            defs = Definitions(
+                assets=[my_table],
+                resources={
+                    "bigquery": BigQueryResource(project="my-project")
+                }
+            )
+    """
+
+    project: Optional[str] = Field(
+        default=None,
+        description="""Project ID for the project which the client acts on behalf of. Will be passed
+        when creating a dataset / job. If not passed, falls back to the default inferred from the
+        environment.""",
```

Comment:
nit:  I think using the triple quote string will add the indents after the newlines here

```suggestion
        default=None,
        description=(
            "Project ID for the project which the client acts on behalf of. Will be passed"
            " when creating a dataset / job. If not passed, falls back to the default inferred from the"
            " environment."
        ),
```